agent/src/runtime_manager.rs

```diff
@@ -108,10 +106,39 @@ impl RuntimeManager {
         }
     }
 
+    // [impl->swdd~agent-initial-list-existing-workloads~1]
+    pub async fn handle_server_hello(
+        &mut self,
+        mut added_workloads: Vec<WorkloadSpec>,
+        workload_state_db: &WorkloadStateStore,
+    ) {
+        log::info!(
+            "Received the server hello with '{}' added workloads.",
+            added_workloads.len()
+        );
+
+        added_workloads = self
+            .resume_and_remove_from_added_workloads(added_workloads)
+            .await;
+
+        let workload_operations: Vec<WorkloadOperation> =
+            self.transform_into_workload_operations(added_workloads, vec![]);
+
+        // [impl->swdd~agent-handles-new-workload-operations]
+        // [impl->swdd~agent-handles-workloads-with-fulfilled-dependencies~1]
+        let ready_workload_operations = self
+            .workload_queue
+            .enqueue_filtered_workload_operations(workload_operations, workload_state_db)
+            .await;
+
+        self.execute_workload_operations(ready_workload_operations)
+            .await;
```

Comment:
We can maybe extract these lines to new function, as we now have duplicated code.

Reply:
Done.